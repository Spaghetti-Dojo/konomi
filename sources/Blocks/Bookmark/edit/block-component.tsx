/**
 * External dependencies
 */
import { SvgBookmark } from '@konomi/icons';
import type { JSX } from 'react';
import React from 'react';

/**
 * WordPress dependencies
 */
// @ts-expect-error
import { useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import type { BookmarkEdit } from './types';

export function BlockComponent(
	props: Readonly< BookmarkEdit.EditProps >
): JSX.Element {
	const { inactiveColor, activeColor } = props.attributes;

	const style: BookmarkEdit.CustomCSSProperties = {
		'--konomi-color--inactive': inactiveColor,
		'--konomi-color--active': activeColor,
	};

	return (
		<div className="konomi-bookmark" style={ style }>
			<button { ...useBlockProps() }>
				<SvgBookmark />
			</button>
		</div>
	);
}
