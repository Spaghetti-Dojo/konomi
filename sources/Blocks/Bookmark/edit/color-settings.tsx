import {
	InspectorControls as _InspectControls,
	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
	__experimentalPanelColorGradientSettings as PanelColorGradientSettings,
	// @ts-expect-error
} from '@wordpress/block-editor';
import { __ } from '@wordpress/i18n';
import type { JSX } from 'react';
import React from 'react';
import type { BookmarkEdit } from './types';

type ColorSetting = Readonly< {
	label: string;
	colorValue: BookmarkEdit.Color;
	onColorChange: ( color: BookmarkEdit.Color ) => void;
	onColorCleared: ( color: BookmarkEdit.Color ) => void;
} >;

// TODO How can I configure the PanelColorGradientSettings to only accept HEX colors?
export function ColorSettings( props: BookmarkEdit.EditProps ): JSX.Element {
	const { inactiveColor, activeColor } = props.attributes;

	const items = [
		createColorSetting(
			__( 'Inactive', 'konomi' ),
			[ 'inactiveColor', inactiveColor ],
			props.setAttributes
		),
		createColorSetting(
			__( 'Active', 'konomi' ),
			[ 'activeColor', activeColor ],
			props.setAttributes
		),
	];

	return (
		<_InspectControls group="styles">
			<PanelColorGradientSettings settings={ items } />
		</_InspectControls>
	);
}

function createColorSetting(
	label: string,
	attribute: Readonly<
		[ 'inactiveColor' | 'activeColor', BookmarkEdit.Color ]
	>,
	onChangeColor: BookmarkEdit.EditProps[ 'setAttributes' ]
): ColorSetting {
	return {
		label,
		colorValue: attribute[ 1 ],
		onColorChange: ( color: BookmarkEdit.Color ): void => {
			onChangeColor( { [ attribute[ 0 ] ]: color } );
		},
		onColorCleared: (): void => {
			onChangeColor( { [ attribute[ 0 ] ]: undefined } );
		},
	};
}
