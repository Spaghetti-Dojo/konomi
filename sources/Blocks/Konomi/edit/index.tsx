/**
 * External dependencies
 */
import type { JSX } from 'react';
import React from 'react';

/**
 * WordPress dependencies
 */
// @ts-expect-error
import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';

/**
 * Internal Dependencies
 */
import type { KonomiEdit } from './types';

export function Edit( props: KonomiEdit.EditProps ): JSX.Element {
	const { attributes } = props;
	const blockProps = useBlockProps();
	const innerBlockProps = useInnerBlocksProps( blockProps, {
		template: [
			[
				'core/group',
				{
					layout: {
						type: 'flex',
						flexWrap: 'nowrap',
					},
				},
				[
					[ 'konomi/reaction', {} ],
					[ 'konomi/bookmark', {} ],
				],
			],
		],
		templateLock: 'insert',
		allowedBlocks: attributes.allowedBlocks,
	} );

	return <div { ...innerBlockProps } />;
}
