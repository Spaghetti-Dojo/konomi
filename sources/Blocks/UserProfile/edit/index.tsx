/**
 * External dependencies
 */
import type { JSX } from 'react';
import React from 'react';
import { SvgHeart, SvgBookmark } from '@konomi/icons';

/**
 * WordPress dependencies
 */
// @ts-expect-error
import { useBlockProps } from '@wordpress/block-editor';

export function Edit(): JSX.Element {
	return (
		<div { ...useBlockProps() }>
			<Table />
		</div>
	);
}

function Table(): JSX.Element {
	return (
		<figure className="wp-block-table is-style-stripes alignwide wp-block-konomi-user-profile">
			<table className="has-fixed-layout">
				<Thead />
				<TBody />
			</table>
		</figure>
	);
}

function Thead(): JSX.Element {
	return (
		<thead>
			<tr>
				<th className="has-text-align-left">Title</th>
				<th className="has-text-align-left">Excerpt</th>
				<th className="has-text-align-center">Actions</th>
			</tr>
		</thead>
	);
}

function TBody(): JSX.Element {
	return (
		<tbody>
			<TRow />
			<TRow />
			<TRow />
		</tbody>
	);
}

function TRow(): JSX.Element {
	return (
		<tr className="konomi-user-profile-item">
			<td className="konomi-user-profile-item__title">
				<a href="https://sample.com">Sample Page</a>
			</td>

			<td className="konomi-user-profile-item__excerpt">
				This is an example page. It’s different from a blog post because
				it will stay…
			</td>

			<td className="konomi-user-profile-item__actions has-text-align-center">
				<div className="konomi">
					<GroupBlock />
				</div>
			</td>
		</tr>
	);
}

function GroupBlock(): JSX.Element {
	return (
		<div
			className="wp-block-group is-nowrap is-layout-flex wp-block-group-is-layout-flex"
			style={ { justifyContent: 'center' } }
		>
			<Reaction />
			<Bookmark />
		</div>
	);
}

function Reaction(): JSX.Element {
	return (
		<div className="konomi-reaction">
			<button className="wp-block-konomi-reaction is-active">
				<SvgHeart />
				<span className="konomi-label screen-reader-text">
					Save this post
				</span>
			</button>
		</div>
	);
}

function Bookmark(): JSX.Element {
	return (
		<div className="konomi-bookmark">
			<button className="wp-block-konomi-bookmark is-active">
				<SvgBookmark />
				<span className="konomi-label screen-reader-text">
					Bookmark this post
				</span>
			</button>
		</div>
	);
}
