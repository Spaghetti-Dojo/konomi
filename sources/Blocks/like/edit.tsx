/**
 * External dependencies
 */
import type { JSX } from 'react';
import React from 'react';
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';
/**
 * External dependencies
 */
import { useConfiguration } from '@konomi/configuration';

export default function Edit(): JSX.Element {
	const { iconsPathUrl } = useConfiguration();

	return (
		<button { ...useBlockProps() }>
			<img
				src={ `${ iconsPathUrl.toString() }/suit-heart.svg` }
				alt="Like"
			/>
		</button>
	);
}
