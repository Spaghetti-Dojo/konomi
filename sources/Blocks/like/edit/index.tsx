/**
 * External dependencies
 */
import { Icon } from '@konomi/icons';
import type { JSX } from 'react';
import React from 'react';
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';

export function Edit(): JSX.Element {
	return (
		<button { ...useBlockProps() }>
			<Icon icon="heart" />
		</button>
	);
}
