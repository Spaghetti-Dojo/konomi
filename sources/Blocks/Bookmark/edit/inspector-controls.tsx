import type { JSX } from 'react';
import React from 'react';
import { ColorSettings } from './color-settings';
import type { BookmarkEdit } from './types';

export function InspectorControls(
	props: BookmarkEdit.EditProps
): JSX.Element {
	return <ColorSettings { ...props } />;
}
