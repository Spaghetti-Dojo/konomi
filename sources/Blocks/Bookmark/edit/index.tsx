/**
 * External dependencies
 */
import type { JSX } from 'react';
import React from 'react';

/**
 * Internal dependencies
 */
import type { BookmarkEdit } from './types';
import { BlockComponent } from './block-component';
import { InspectorControls } from './inspector-controls';

export function Edit( props: Readonly< BookmarkEdit.EditProps > ): JSX.Element {
	return (
		<>
			<BlockComponent { ...props } />
			<InspectorControls { ...props } />
		</>
	);
}
