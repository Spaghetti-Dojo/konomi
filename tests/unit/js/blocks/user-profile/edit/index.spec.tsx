import React from 'react';
import { jest, describe, it, expect } from '@jest/globals';
import { render } from '@testing-library/react';
import { Edit } from '../../../../../../sources/Blocks/UserProfile/edit/index';

jest.mock( '@wordpress/block-editor', () => ( {
	// @ts-ignore
	useBlockProps: ( ...props ) => ( {
		className: 'mock-block-props',
		...props,
	} ),
} ) );

describe( 'BlockComponent', () => {
	it( 'should match snapshot', () => {
		const props = {}
		const { container } = render( <Edit { ...props } /> );
		expect( container ).toMatchSnapshot();
	} );
} );
