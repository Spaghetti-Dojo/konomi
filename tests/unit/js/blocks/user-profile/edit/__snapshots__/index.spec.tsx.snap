// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockComponent should match snapshot 1`] = `
<div>
  <div
    class="mock-block-props"
  >
    <figure
      class="wp-block-table is-style-stripes alignwide wp-block-konomi-user-profile"
    >
      <table
        class="has-fixed-layout"
      >
        <thead>
          <tr>
            <th
              class="has-text-align-left"
            >
              Title
            </th>
            <th
              class="has-text-align-left"
            >
              Excerpt
            </th>
            <th
              class="has-text-align-center"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="konomi-user-profile-item"
          >
            <td
              class="konomi-user-profile-item__title"
            >
              <a
                href="https://sample.com"
              >
                Sample Page
              </a>
            </td>
            <td
              class="konomi-user-profile-item__excerpt"
            >
              This is an example page. It’s different from a blog post because it will stay…
            </td>
            <td
              class="konomi-user-profile-item__actions has-text-align-center"
            >
              <div
                class="konomi"
              >
                <div
                  class="wp-block-group is-nowrap is-layout-flex wp-block-group-is-layout-flex"
                  style="justify-content: center;"
                >
                  <div
                    class="konomi-reaction"
                  >
                    <button
                      class="wp-block-konomi-reaction is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Save this post
                      </span>
                    </button>
                  </div>
                  <div
                    class="konomi-bookmark"
                  >
                    <button
                      class="wp-block-konomi-bookmark is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Bookmark this post
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="konomi-user-profile-item"
          >
            <td
              class="konomi-user-profile-item__title"
            >
              <a
                href="https://sample.com"
              >
                Sample Page
              </a>
            </td>
            <td
              class="konomi-user-profile-item__excerpt"
            >
              This is an example page. It’s different from a blog post because it will stay…
            </td>
            <td
              class="konomi-user-profile-item__actions has-text-align-center"
            >
              <div
                class="konomi"
              >
                <div
                  class="wp-block-group is-nowrap is-layout-flex wp-block-group-is-layout-flex"
                  style="justify-content: center;"
                >
                  <div
                    class="konomi-reaction"
                  >
                    <button
                      class="wp-block-konomi-reaction is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Save this post
                      </span>
                    </button>
                  </div>
                  <div
                    class="konomi-bookmark"
                  >
                    <button
                      class="wp-block-konomi-bookmark is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Bookmark this post
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="konomi-user-profile-item"
          >
            <td
              class="konomi-user-profile-item__title"
            >
              <a
                href="https://sample.com"
              >
                Sample Page
              </a>
            </td>
            <td
              class="konomi-user-profile-item__excerpt"
            >
              This is an example page. It’s different from a blog post because it will stay…
            </td>
            <td
              class="konomi-user-profile-item__actions has-text-align-center"
            >
              <div
                class="konomi"
              >
                <div
                  class="wp-block-group is-nowrap is-layout-flex wp-block-group-is-layout-flex"
                  style="justify-content: center;"
                >
                  <div
                    class="konomi-reaction"
                  >
                    <button
                      class="wp-block-konomi-reaction is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Save this post
                      </span>
                    </button>
                  </div>
                  <div
                    class="konomi-bookmark"
                  >
                    <button
                      class="wp-block-konomi-bookmark is-active"
                    >
                      <svg />
                      <span
                        class="konomi-label screen-reader-text"
                      >
                        Bookmark this post
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </figure>
  </div>
</div>
`;
